### DISCUSSION TITLE
This is not going well.

### DISCUSSION THREAD

**Discussion by richardstevenhack** [nicobrauchtgit/obsidian-Smart2Brain#146] [Q&A]
Running latest version on openSUSE Tumbleweed Linux on a Ryzen 9 5950X with 128GB RAM, and a mere 2GB old AMD Radeon  GPU (so CPU-only). Running Ollama with Qwen2 LLM and nomic-embedding LLM.

I have around 3,000 notes in the Vault. It took S2B about four hours to index the files. I wasn't watching when it finished, but it came back with the chat panel, so I assume it finished.

So I tried to get it to answer a question about a topic in my vault. The first time it gave an error but the error did not remain on the screen, so I have no idea what it was. Is there a log somewhere that I can view?

Next, I asked it if it could see the specific folder that holds the files on the topic in question. It then said: "Reducing 33 Notes." After it had been doing that for five or ten minutes, I attempted a request to list the files in that folder, but it said it was still processing and to please wait. I've now been waiting another ten minutes.

So far this is not usable. If I could see a log of its activity somewhere - the Obsidian Console doesn't show anything useful relative to S2B, just some messages from my earlier interactions with Text Generator.

Any help would be appreciated.

UPDATE: It finally finished the reducing 33 files, then gave me a reply which has nothing to do with listing the files but was answering the earlier question as to reading the folder but in a generic "how to use Obsidian" format which was completely unresponsive.

So now I asked it whether IT can read the file folder. Again, I got the error message which starts with something like "Unable to start Smart2Brain". The rest of the error message disappears too fast to see.

Ah, I did a screen capture fast enough to catch the error message. This is what it says:

"Failed to run SmartSecondBrain. [Error:,Error User query is too long or a single document was longer than the context length (should not happen as we split documents by length in post processing.) Please retry."

So what I will do now is try a very small document...

So now I asked it to summarize a document which has ten sentences total in it. It then went back to "Reducing 36 Notes"! 

I just turned on Developer Console Logging in the settings, so maybe I'll get more info.

The response to the question about the ten sentence document was hilarious. It took another 10-15 minutes, and then responded with a weird summary of everything in the vault!

Note that my Creativity setting is 70% - which I suspect is way too high - and Similarity is at 50% which I suspect is way too low. So let me adjust that...

So I adjusted the Creativity down to 30% and the Similarity to 75% and asked it to specifically summarize the one document. Instead it said "No documents retrieved" and suggested reducing the Similarity level.

This thing is unusable at this point.

**Comment by richardstevenhack** [comment]
So it finally comes back with a response to the request to summarize that one document as follows:

"To provide a summary of the Vault Note titled "Reality", please refer to the Obsidian notes section where this specific note is stored. This note likely delves into discussions or insights about reality, possibly exploring philosophical concepts, personal experiences, or scientific perspectives on the nature of existence and perception. For an accurate summary, navigate through your Obsidian vault using the search function or directly access the "Reality" note to gather its content. [Vault Note: Reality](app://obsidian.md/Vault Note:Reality)" 

This note I was referencing is in the top level portion of the vault; it's not in a folder. S2B/Gwen2 can't even find it.

The latter words "Vault Note: Reality" were in a link - which can't work because Obsidian does not allow special characters in note titles.

I have to say I was moderately impressed when it indexed my whole vault (supposedly, I don't have any proof it actually did) within four hours, as I was concerned how long that would take. But as it stands, this is utterly unusable except as a chatbot with the LLM. I already have MSTY installed which can do that as well as index the vault (although I don't know how long that will take as I haven't tried it yet) as well as do RAQ with individual files and folders.

I hope someone has some ideas on how to make this usable because it's one of the few utilities that claims to do RAG with an Obsidian vault.

**Comment by Leo310** [comment]
I know Its been some time, sorry for that, but it should be fixed in the next Release ;)
Wont take long anymore!

**Reply by richardstevenhack** [reply]
Unfortunately I am well past using your utility. I pretty much do everything now with MSTY Studio.
